gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/SSOCookieFederationFilter.java

```diff
@@ -93,6 +95,13 @@ public void init( FilterConfig filterConfig ) throws ServletException {
 
     // gateway path for deriving an idp url when missing
     gatewayPath = filterConfig.getInitParameter(GATEWAY_PATH);
+    if (StringUtils.isBlank(gatewayPath)) {
+      final GatewayConfig gatewayConfig = filterConfig.getServletContext() == null ? null
+          : (GatewayConfig) filterConfig.getServletContext().getAttribute(GatewayConfig.GATEWAY_CONFIG_ATTRIBUTE);
+      if (gatewayConfig != null) {
+        gatewayPath = gatewayConfig.getGatewayPath();
+      }
+    }
```

Comment:
This should be its own method

Reply:
Do we need `StringUtils.isBlank`? Check is not null and is not empty? That would remove the commons-lang3 dependency

Reply:
I know you are not fan of adding new dependencies but the `commons-lang` dependency is in our project anyway. If you are insisting on removing it I'll do...

Reply:
> This should be its own method

+1 (let me wait a bit until for review comments before submitting a new patchset)

Reply:
Fixed:
- setting `gatewayPath` in its own method
- removed the `commons-lang3` dependency